### PR TITLE
COMP: address unused parameter warning in default constructor

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist

-  Makes breaking changes to the vxl/core/\* API that requires semantic versioning increase
-  Makes design changes to existing vxl/core\* API that requires semantic versioning increase

-  Makes changes to the contributed directory API DOES NOT require semantic versioning increase
-  Adds tests and baseline comparison (quantitative).
-  Adds Documentation.

### CODE REVIEW

**Comment by blowekamp** [comment]
@hjmjohnson 
Addresses InsightSoftwareConsortium/ITK#1430

File: core/vnl/vnl_vector.hxx [hunk 1]
```diff
@@ -105,7 +105,9 @@ vnl_vector<T>::vnl_vector (size_t len, size_t n, T const values[])
 {
   vnl_vector_alloc_blah(len);
   // If user specified values, initialize first n elements with values
-  const size_t copy_num{std::min(len,n) };
+  // n.b assignment is used over universal initialization for avoid a
+  // gcc 4.8.5 ICE
+  const size_t copy_num = std::min(len,n);
```
Comment (hjmjohnson) [inline core/vnl/vnl_vector.hxx]: I finished building vxl in the docker container you provided at 2:06pm :) 2 min before this arrived.    I made the exact same change!

Hans

File: core/vnl/vnl_matrix_ref.h [hunk 2]
```diff
@@ -50,7 +50,7 @@ class VNL_EXPORT vnl_matrix_ref : public vnl_matrix<T>
   { }
 
   //vnl_matrix base class is not no_except, so derived class can not be either
-  vnl_matrix_ref(vnl_matrix_ref<T> && rhs) = default;
+  vnl_matrix_ref(vnl_matrix_ref<T> &&) = default;
```
Comment (hjmjohnson) [inline core/vnl/vnl_matrix_ref.h]: Nice! Thanks.

**Comment by blowekamp** [comment]
I have the PR correct now with just one commit.

**Comment by hjmjohnson** [comment]
@blowekamp  Let's wait ~18 minutes, and I'll merge once it is returned with the anticipated green.

**Comment by dzenanz** [comment]
Time to merge?

**Comment by hjmjohnson** [comment]
@dzenanz Merged!

**Pull request merged**